libminifi/src/io/tls/TLSServerSocket.cpp

```diff
@@ -87,13 +87,13 @@ void TLSServerSocket::registerCallback(std::function<bool()> accept_function, st
  * Initializes the socket
  * @return result of the creation operation.
  */
-void TLSServerSocket::registerCallback(std::function<bool()> accept_function, std::function<int(std::vector<uint8_t>*, int *)> handler) {
-  fx = [this](std::function<bool()> accept_function, std::function<int(std::vector<uint8_t>*, int *)> handler) {
+void TLSServerSocket::registerCallback(std::function<bool()> accept_function, std::function<int(std::vector<uint8_t>*, int *)> handler, std::chrono::milliseconds timeout) {
+  fx = [this](std::function<bool()> accept_function, std::function<int(std::vector<uint8_t>*, int *)> handler, std::chrono::milliseconds timeout) {
     int ret = 0;
     std::vector<int> fds;
     int size;
     while (accept_function()) {
-      int fd = select_descriptor(3000);
+      int fd = select_descriptor(timeout.count());
```

Comment:
this is a narrowing conversion from at least 45 bits to 16 bits; could be made safe (and warning-free) with `gsl::narrow`

Reply:
Fixed in [bfc8394](https://github.com/apache/nifi-minifi-cpp/pull/959/commits/bfc83949f4c3549d5a85fc0250068eea7e7801d1)